htlcswitch/link.go

```diff
@@ -1726,15 +1726,19 @@ func (l *channelLink) processLockedInHtlcs(
 					fwdInfo.AmountToForward,
 				)
 
-				// If the amount of the incoming HTLC, minus
-				// our expected fee isn't equal to the
-				// forwarding instructions, then either the
-				// values have been tampered with, or the send
-				// used incorrect/dated information to
+				// If the actual fee is less than our expected
+				// fee, then we'll reject this HTLC as it
+				// didn't provide a sufficient amount of fees,
+				// or the values have been tampered with, or
+				// the send used incorrect/dated information to
 				// construct the forwarding information for
 				// this hop. In any case, we'll cancel this
 				// HTLC.
-				if pd.Amount-expectedFee < fwdInfo.AmountToForward {
+				//
+				// TODO(roasbeef): more aggressive consistency checks
+				actualFee := pd.Amount - fwdInfo.AmountToForward
```

Comment:
are we always sure that `pd.Amount >= fwdInfo.AmountToForward` at this point?

Reply:
Nope, added the check, and resolved the TODO in the process ;)

Reply:
If `pd.Amount == fwdInfo.AmountToForward`, the I believe this will fail?